src/components/List/ApplicationsList.tsx

```diff
@@ -95,13 +98,34 @@ const ApplicationRow: React.FC<ApplicationRowProps> = ({ refetch, columns, appli
 
   if (error) return <Message error header={t('ERROR_APPLICATION_DELETE')} list={[error]} />
 
+  const MobileActionCell = getMobileActionCell(isMobile, columns[ACTION_CELL_INDEX], application)
+
   return (
-    <Table.Row key={`ApplicationList-application-${application.id}`} className="list-row">
+    <Table.Row
+      key={`ApplicationList-application-${application.id}`}
+      className="list-row"
+      style={{ position: 'relative' }}
+    >
       {columns.map(({ ColumnComponent, headerName, hideMobileLabel, hideOnMobileTest }, index) => {
+        if (index === ACTION_CELL_INDEX && MobileActionCell)
+          return (
+            <Table.Cell
+              key={`ApplicationList-row-${application.id}-action`}
+              className="mobile-action-cell"
+              style={{ position: 'absolute', top: '1em', right: '1em' }}
+            >
+              <MobileActionCell {...props} />
+            </Table.Cell>
+          )
```

Comment:
For the reason mentioned above, we have to do some additional logic to even determine if there's going to be an Action cell for this row. And if there is, we render it last (index == 6) as a special case (`MobileActionCell`), but if not, we just render the cells as before -- wrapping in `TableCellMobileLabelWrapper` and hiding/formatting based on definitions in `mapColumns...`